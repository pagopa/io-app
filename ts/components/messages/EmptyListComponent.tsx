import { Text, View } from "native-base";
import * as React from "react";
import { Image, ImageSourcePropType, StyleSheet } from "react-native";
import customVariables from "../../theme/variables";

const styles = StyleSheet.create({
  view: {
    padding: customVariables.contentPadding,
    alignItems: "center"
  },
  title: {
    paddingTop: customVariables.contentPadding
  },
  subtitle: {
    textAlign: "center",
    paddingTop: customVariables.contentPadding,
    fontSize: customVariables.fontSizeSmall
  }
});

type Props = Readonly<{
  image: ImageSourcePropType;
  title: string;
  subtitle?: string;
}>;

export const EmptyListComponent = (props: Props) => {
  return (
    <View style={styles.view}>
      <View spacer={true} />
      <Image source={props.image} />
      <Text style={styles.title}>{props.title}</Text>
      {props.subtitle && <Text style={styles.subtitle}>{props.subtitle}</Text>}
    </View>
  );
};
