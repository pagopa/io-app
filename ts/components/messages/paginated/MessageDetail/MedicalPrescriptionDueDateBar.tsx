import { capitalize } from "lodash";
import { Text, View } from "native-base";
import React from "react";
import { StyleSheet } from "react-native";

import I18n from "../../../../i18n";
import customVariables from "../../../../theme/variables";
import {
  format,
  formatDateAsDay,
  formatDateAsLocal,
  formatDateAsMonth
} from "../../../../utils/dates";
import {
  isExpired,
  isExpiring,
  MessagePaymentExpirationInfo
} from "../../../../utils/messages";
import { UIMessageDetails } from "../../../../store/reducers/entities/messages/types";
import CalendarIconComponent from "../../CalendarIconComponent";
import CalendarEventButton from "../../CalendarEventButton";

type Props = {
  dueDate: Date;

  messageDetails: UIMessageDetails;

  /* A message can carry both medical data and a payment */
  paymentExpirationInfo?: MessagePaymentExpirationInfo;
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: customVariables.contentPadding,
    paddingVertical: customVariables.appHeaderPaddingHorizontal
  },
  text: {
    flex: 1
  },
  row: {
    flexDirection: "row",
    alignItems: "center"
  }
});

/**
 * Show detailed info about the due date of a prescription.
 * Note that payment-related data is used to infer a due date and its 'expire status'.
 *
 * TODO:
 * - update with medical prescription due date (if different from the message due date)
 * - if due date is different, check what to pass to CalendarEventButton
 */
const MedicalPrescriptionDueDateBar = ({
  dueDate,
  messageDetails,
  paymentExpirationInfo
}: Props) => {
  const isPrescriptionExpired = paymentExpirationInfo
    ? isExpired(paymentExpirationInfo)
    : false;
  const isPrescriptionExpiring = paymentExpirationInfo
    ? isExpiring(paymentExpirationInfo)
    : false;

  const date = formatDateAsLocal(dueDate, true, true);
  const time = format(dueDate, "HH.mm");

  const bannerStyle = isPrescriptionExpired
    ? { backgroundColor: customVariables.brandDarkGray }
    : isPrescriptionExpiring
    ? { backgroundColor: customVariables.calendarExpirableColor }
    : { backgroundColor: customVariables.brandGray };

  if (isPrescriptionExpiring || isPrescriptionExpired) {
    return (
      <View
        testID={"MedicalPrescriptionDueDate_expired_or_expiring"}
        style={[styles.container, styles.row, bannerStyle]}
      >
        <CalendarIconComponent
          month={capitalize(formatDateAsMonth(dueDate))}
          day={formatDateAsDay(dueDate)}
          backgroundColor={customVariables.colorWhite}
          textColor={
            isPrescriptionExpiring
              ? customVariables.calendarExpirableColor
              : customVariables.brandDarkGray
          }
        />

        <View hspacer={true} small={true} />
        <Text style={styles.text} white={true}>
          {isPrescriptionExpiring &&
            I18n.t("messages.cta.prescription.expiringAlert")}
          {isPrescriptionExpired && (
            <>
              {I18n.t("messages.cta.prescription.expiredAlert.block1")}
              <Text bold={true} white={true}>{` ${time} `}</Text>
              {I18n.t("messages.cta.prescription.expiredAlert.block2")}
              <Text bold={true} white={true}>{` ${date}`}</Text>
            </>
          )}
        </Text>
      </View>
    );
  }

  return (
    <View
      testID={"MedicalPrescriptionDueDate_valid"}
      style={[styles.container, bannerStyle]}
    >
      <Text style={styles.text} white={false}>
        {I18n.t("messages.cta.prescription.addMemo")}
        <Text bold={true}>{` ${date}`}</Text>
      </Text>
      <View spacer={true} xsmall={true} />
      <View style={styles.row}>
        <CalendarIconComponent
          month={capitalize(formatDateAsMonth(dueDate))}
          day={formatDateAsDay(dueDate)}
          backgroundColor={customVariables.brandDarkGray}
          textColor={customVariables.colorWhite}
        />
        <View hspacer={true} small={true} />
        <CalendarEventButton message={messageDetails.raw} medium={true} />
      </View>
    </View>
  );
};

export default MedicalPrescriptionDueDateBar;
