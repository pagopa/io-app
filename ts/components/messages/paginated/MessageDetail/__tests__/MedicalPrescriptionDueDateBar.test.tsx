import React from "react";
import configureMockStore from "redux-mock-store";
import { NavigationParams } from "react-navigation";

import MedicalPrescriptionDueDateBar from "../MedicalPrescriptionDueDateBar";
import { toUIMessageDetails } from "../../../../../store/reducers/entities/messages/transformers";
import { paymentValidInvalidAfterDueDate } from "../../../../../__mocks__/message";
import { appReducer } from "../../../../../store/reducers";
import { applicationChangeState } from "../../../../../store/actions/application";
import { GlobalState } from "../../../../../store/reducers/types";
import { renderScreenFakeNavRedux } from "../../../../../utils/testWrapper";
import ROUTES from "../../../../../navigation/routes";
import { MessagePaymentExpirationInfo } from "../../../../../utils/messages";

jest.useFakeTimers();

const uiMessageDetails = toUIMessageDetails(paymentValidInvalidAfterDueDate);
const dueDate = uiMessageDetails.dueDate!; // new Date("2021-11-17T12:32:14.497Z");
const defaultProps: React.ComponentProps<typeof MedicalPrescriptionDueDateBar> =
  {
    dueDate,
    messageDetails: uiMessageDetails,
    paymentExpirationInfo: {
      kind: "EXPIRABLE",
      expireStatus: "EXPIRED",
      dueDate
    }
  };

describe("MedicalPrescriptionDueDateBar component", () => {
  describe("when payment info is expired", () => {
    it("should match the snapshot", () => {
      expect(
        renderComponent(defaultProps).component.toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when payment info is expiring", () => {
    it("should match the snapshot", () => {
      expect(
        renderComponent({
          ...defaultProps,
          paymentExpirationInfo: {
            ...defaultProps.paymentExpirationInfo,
            expireStatus: "EXPIRING"
          } as MessagePaymentExpirationInfo
        }).component.toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when payment info is valid", () => {
    it("should match the snapshot", () => {
      expect(
        renderComponent({
          ...defaultProps,
          paymentExpirationInfo: {
            ...defaultProps.paymentExpirationInfo,
            expireStatus: "VALID"
          } as MessagePaymentExpirationInfo
        }).component.toJSON()
      ).toMatchSnapshot();
    });
  });

  describe("when payment info is not expirable", () => {
    it("should match the snapshot", () => {
      expect(
        renderComponent({
          ...defaultProps,
          paymentExpirationInfo: {
            kind: "UNEXPIRABLE"
          }
        }).component.toJSON()
      ).toMatchSnapshot();
    });
  });
});

const renderComponent = (
  props: React.ComponentProps<typeof MedicalPrescriptionDueDateBar>
) => {
  const globalState = appReducer(undefined, applicationChangeState("active"));
  const mockStore = configureMockStore<GlobalState>();
  const store: ReturnType<typeof mockStore> = mockStore(globalState);

  return {
    component: renderScreenFakeNavRedux<GlobalState, NavigationParams>(
      () => <MedicalPrescriptionDueDateBar {...props} />,
      ROUTES.MESSAGE_DETAIL,
      {},
      store
    ),
    store
  };
};
