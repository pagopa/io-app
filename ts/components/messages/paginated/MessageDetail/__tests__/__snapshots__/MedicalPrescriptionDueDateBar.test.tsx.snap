// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicalPrescriptionDueDateBar component when payment info is expired should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        forwardedRef={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 1000000,
              },
              Object {
                "translateY": 1000000,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        forwardedRef={[Function]}
        pointerEvents="box-only"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "#475A6D",
                },
              ]
            }
            testID="MedicalPrescriptionDueDate_expired_or_expiring"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={40}
              bbWidth={40}
              focusable={false}
              height={40}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
              vbHeight={40}
              vbWidth={40}
              width={40}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  name="calendar"
                >
                  <RNSVGPath
                    d="m 33,0 h 3 c 2.209139,0 4,1.790861 4,4 v 32 c 0,2.209139 -1.790861,4 -4,4 H 4 C 1.790861,40 0,38.209139 0,36 V 4 C 0,1.790861 1.790861,0 4,0 h 3 v 3 h 4 V 0 h 18 v 3 h 4 z"
                    fill={4294967295}
                    name="icon"
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGText
                    alignmentBaseline="central"
                    content={null}
                    dx={Array []}
                    dy={Array []}
                    fill={4282866285}
                    font={
                      Object {
                        "fontSize": 10,
                        "textAnchor": "middle",
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotate={Array []}
                    x={
                      Array [
                        "50%",
                      ]
                    }
                    y={
                      Array [
                        "30%",
                      ]
                    }
                  >
                    <RNSVGTSpan
                      content="Dec"
                      dx={Array []}
                      dy={Array []}
                      font={Object {}}
                      rotate={Array []}
                      x={Array []}
                      y={Array []}
                    />
                  </RNSVGText>
                  <RNSVGText
                    alignmentBaseline="central"
                    content={null}
                    dx={Array []}
                    dy={Array []}
                    fill={4282866285}
                    font={
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAnchor": "middle",
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotate={Array []}
                    x={
                      Array [
                        "50%",
                      ]
                    }
                    y={
                      Array [
                        "65%",
                      ]
                    }
                  >
                    <RNSVGTSpan
                      content="01"
                      dx={Array []}
                      dy={Array []}
                      font={Object {}}
                      rotate={Array []}
                      x={Array []}
                      y={Array []}
                    />
                  </RNSVGText>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              hspacer={true}
              small={true}
              style={Object {}}
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              white={true}
            >
              The prescription expired at
              <Text
                bold={true}
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                white={true}
              >
                 13.29 
              </Text>
              on
              <Text
                bold={true}
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                white={true}
              >
                 12/01/2021
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`MedicalPrescriptionDueDateBar component when payment info is expiring should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        forwardedRef={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 1000000,
              },
              Object {
                "translateY": 1000000,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        forwardedRef={[Function]}
        pointerEvents="box-only"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "#D0021B",
                },
              ]
            }
            testID="MedicalPrescriptionDueDate_expired_or_expiring"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={40}
              bbWidth={40}
              focusable={false}
              height={40}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
              vbHeight={40}
              vbWidth={40}
              width={40}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  name="calendar"
                >
                  <RNSVGPath
                    d="m 33,0 h 3 c 2.209139,0 4,1.790861 4,4 v 32 c 0,2.209139 -1.790861,4 -4,4 H 4 C 1.790861,40 0,38.209139 0,36 V 4 C 0,1.790861 1.790861,0 4,0 h 3 v 3 h 4 V 0 h 18 v 3 h 4 z"
                    fill={4294967295}
                    name="icon"
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                  <RNSVGText
                    alignmentBaseline="central"
                    content={null}
                    dx={Array []}
                    dy={Array []}
                    fill={4291822107}
                    font={
                      Object {
                        "fontSize": 10,
                        "textAnchor": "middle",
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotate={Array []}
                    x={
                      Array [
                        "50%",
                      ]
                    }
                    y={
                      Array [
                        "30%",
                      ]
                    }
                  >
                    <RNSVGTSpan
                      content="Dec"
                      dx={Array []}
                      dy={Array []}
                      font={Object {}}
                      rotate={Array []}
                      x={Array []}
                      y={Array []}
                    />
                  </RNSVGText>
                  <RNSVGText
                    alignmentBaseline="central"
                    content={null}
                    dx={Array []}
                    dy={Array []}
                    fill={4291822107}
                    font={
                      Object {
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAnchor": "middle",
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotate={Array []}
                    x={
                      Array [
                        "50%",
                      ]
                    }
                    y={
                      Array [
                        "65%",
                      ]
                    }
                  >
                    <RNSVGTSpan
                      content="01"
                      dx={Array []}
                      dy={Array []}
                      font={Object {}}
                      rotate={Array []}
                      x={Array []}
                      y={Array []}
                    />
                  </RNSVGText>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
            <View
              hspacer={true}
              small={true}
              style={Object {}}
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              white={true}
            >
              This prescription expires today
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`MedicalPrescriptionDueDateBar component when payment info is not expirable should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        forwardedRef={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 1000000,
              },
              Object {
                "translateY": 1000000,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        forwardedRef={[Function]}
        pointerEvents="box-only"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                },
                Object {
                  "backgroundColor": "#F5F6F7",
                },
              ]
            }
            testID="MedicalPrescriptionDueDate_valid"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              white={false}
            >
              This prescription expires the
              <Text
                bold={true}
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
              >
                 12/01/2021
              </Text>
            </Text>
            <View
              spacer={true}
              style={Object {}}
              xsmall={true}
            />
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={40}
                bbWidth={40}
                focusable={false}
                height={40}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
                vbHeight={40}
                vbWidth={40}
                width={40}
              >
                <RNSVGGroup>
                  <RNSVGGroup
                    name="calendar"
                  >
                    <RNSVGPath
                      d="m 33,0 h 3 c 2.209139,0 4,1.790861 4,4 v 32 c 0,2.209139 -1.790861,4 -4,4 H 4 C 1.790861,40 0,38.209139 0,36 V 4 C 0,1.790861 1.790861,0 4,0 h 3 v 3 h 4 V 0 h 18 v 3 h 4 z"
                      fill={4282866285}
                      name="icon"
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGText
                      alignmentBaseline="central"
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 10,
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "30%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="Dec"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                    <RNSVGText
                      alignmentBaseline="central"
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "65%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="01"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                hspacer={true}
                small={true}
                style={Object {}}
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                forwardedRef={[Function]}
                hitSlop={
                  Object {
                    "bottom": 5,
                    "top": 5,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 1,
                    "borderColor": "#007aff",
                    "borderLeftWidth": 1,
                    "borderRadius": 5,
                    "borderRightWidth": 1,
                    "borderTopWidth": 1,
                    "borderWidth": 1,
                    "elevation": null,
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 30,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessibilityLabel=""
                  accessible={false}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {},
                        Object {},
                        Object {},
                      ],
                      Object {
                        "fontFamily": "io-icon-font",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  $
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#007aff",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "marginTop": 1,
                      },
                    ]
                  }
                >
                  Reminder
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`MedicalPrescriptionDueDateBar component when payment info is valid should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        forwardedRef={[Function]}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 1000000,
              },
              Object {
                "translateY": 1000000,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        forwardedRef={[Function]}
        pointerEvents="box-only"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "paddingHorizontal": 24,
                  "paddingVertical": 12,
                },
                Object {
                  "backgroundColor": "#F5F6F7",
                },
              ]
            }
            testID="MedicalPrescriptionDueDate_valid"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
              white={false}
            >
              This prescription expires the
              <Text
                bold={true}
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
              >
                 12/01/2021
              </Text>
            </Text>
            <View
              spacer={true}
              style={Object {}}
              xsmall={true}
            />
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={40}
                bbWidth={40}
                focusable={false}
                height={40}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
                vbHeight={40}
                vbWidth={40}
                width={40}
              >
                <RNSVGGroup>
                  <RNSVGGroup
                    name="calendar"
                  >
                    <RNSVGPath
                      d="m 33,0 h 3 c 2.209139,0 4,1.790861 4,4 v 32 c 0,2.209139 -1.790861,4 -4,4 H 4 C 1.790861,40 0,38.209139 0,36 V 4 C 0,1.790861 1.790861,0 4,0 h 3 v 3 h 4 V 0 h 18 v 3 h 4 z"
                      fill={4282866285}
                      name="icon"
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGText
                      alignmentBaseline="central"
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 10,
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "30%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="Dec"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                    <RNSVGText
                      alignmentBaseline="central"
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4294967295}
                      font={
                        Object {
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "65%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="01"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                hspacer={true}
                small={true}
                style={Object {}}
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                forwardedRef={[Function]}
                hitSlop={
                  Object {
                    "bottom": 5,
                    "top": 5,
                  }
                }
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "flex-start",
                    "backgroundColor": "transparent",
                    "borderBottomWidth": 1,
                    "borderColor": "#007aff",
                    "borderLeftWidth": 1,
                    "borderRadius": 5,
                    "borderRightWidth": 1,
                    "borderTopWidth": 1,
                    "borderWidth": 1,
                    "elevation": null,
                    "flex": 1,
                    "flexDirection": "row",
                    "height": 30,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessibilityLabel=""
                  accessible={false}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {},
                        Object {},
                        Object {},
                      ],
                      Object {
                        "fontFamily": "io-icon-font",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  $
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#007aff",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                      },
                      Object {
                        "marginTop": 1,
                      },
                    ]
                  }
                >
                  Reminder
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
